packages/core/src/components/Tooltip/index.tsx

```diff
@@ -213,8 +226,52 @@ export class Tooltip extends React.Component<TooltipProps & WithStylesProps, Too
 
   private handleMouseLeave = () => {
     if (!this.props.toggleOnClick) {
-      this.handleClose();
+      if (this.props.popover) {
+        this.delaySetPopupVisible(false, MOUSE_LEAVE_DELAY);
+      } else {
+        this.handleClose();
+      }
+    }
+  };
+
+  setPopupVisible(open: boolean) {
+    const { open: previousOpen } = this.state;
+    this.clearDelayTimer();
+    if (previousOpen !== open) {
+      if (open) {
+        this.handleOpen();
+      } else {
+        this.handleClose();
+      }
+    }
+  }
+
+  clearDelayTimer() {
+    if (this.delayTimer) {
+      clearTimeout(this.delayTimer);
+      this.delayTimer = null;
+    }
+  }
+
+  delaySetPopupVisible(open: boolean, delayS: number) {
+    const delay = delayS * 1000;
+    this.clearDelayTimer();
+    if (delay) {
+      this.delayTimer = window.setTimeout(() => {
+        this.setPopupVisible(open);
+        this.clearDelayTimer();
```

Comment:
I don't think you need to clear this because it will have executed by the time this runs.